tuf/utils/x509_test.go

```diff
@@ -172,17 +193,36 @@ func TestKeyOperations(t *testing.T) {
 
 	// quick test that gun headers are being added appropriately
 	// Encrypt our RSA Key, one type of key should be enough since headers are treated the same
-	testGunKey, err := EncryptPrivateKey(rsaKey, data.CanonicalRootRole, "ilove", "ponies")
+	testGunKey, err := ConvertPrivateKeyToPKCS8(rsaKey, data.CanonicalRootRole, "ilove", "ponies")
 	require.NoError(t, err)
 
-	testNoGunKey, err := EncryptPrivateKey(rsaKey, data.CanonicalRootRole, "", "ponies")
+	testNoGunKey, err := ConvertPrivateKeyToPKCS8(rsaKey, data.CanonicalRootRole, "", "ponies")
 	require.NoError(t, err)
 
-	stringTestGunKey := string(testGunKey)
-	require.True(t, strings.Contains(stringTestGunKey, "gun: ilove"))
+	_, gun, err := ExtractPrivateKeyAttributes(testGunKey)
+	require.NoError(t, err)
+	require.EqualValues(t, "ilove", gun)
+
+	_, gun, err = ExtractPrivateKeyAttributes(testNoGunKey)
+	require.NoError(t, err)
+	require.EqualValues(t, "", gun)
+}
+
+func testKeyBlockType(t *testing.T, b, password []byte, expectedKeyType string) {
+	block, _ := pem.Decode(b)
+
+	var privKey data.PrivateKey
+	var err error
+	if password == nil {
```

Comment:
nit: we don't need this conditional or the variable definitions above.
We should also check that the block isn't nil.

This can just be:
```
block, _ := pem.Decode(b)
require.NotNil(block)
privKey, err := ParsePKCS8ToTufKey(block.Bytes, password)
...
```